compass/landice/tests/ismip6_forcing/ocean_thermal/process_thermal_forcing.py

```diff
@@ -0,0 +1,319 @@
+        args = ["ncremap",
+                "-i", input_file,
+                "-o", output_file,
+                "-m", mapping_file]
+
+        subprocess.check_call(args)
+
+    def rename_ismip6thermalforcing_to_mali_vars(self, remapped_file_temp,
+                                                 output_file):
+        """
+        Rename variables in the remapped ismip6 input data
+        to the ones that MALI uses.
+
+        Parameters
+        ----------
+        remapped_file_temp : str
+            temporary ismip6 data remapped on mali mesh
+        output_file : str
+            remapped ismip6 data renamed on mali mesh
+        """
+
+        config = self.config
+        section = config['ismip6_ais_ocean_thermal']
+        process_obs_data = section.getboolean('process_obs_data')
+
+        # open dataset in 20 years chunk
+        ds = xr.open_dataset(remapped_file_temp, chunks=dict(time=20),
+                             engine="netcdf4")
+
+        ds["ismip6shelfMelt_zOcean"] = ds.z
+        ds = ds.drop_vars('z')  # dropping 'z' while it's still called 'z'
+
+        # build dictionary for ismip6 variables that MALI takes in
+        if process_obs_data:
+            ismip6_to_mali_dims = dict(
+                z="nISMIP6OceanLayers",
+                ncol="nCells")
+            ds["xtime"] = ("Time", ["2015-01-01_00:00:00".ljust(64)])
+            ds["xtime"] = ds.xtime.astype('S')
+            ds["thermal_forcing"] = ds["thermal_forcing"].expand_dims(
+                dim="Time", axis=0)
+            ds = ds.rename(ismip6_to_mali_dims)
+        else:
+            ismip6_to_mali_dims = dict(
+                z="nISMIP6OceanLayers",
+                time="Time",
+                ncol="nCells")
+            ds = ds.rename(ismip6_to_mali_dims)
+            # add xtime variable
+            xtime = []
+            for t_index in range(ds.sizes["Time"]):
+                date = ds.Time[t_index]
+                # forcing files do not all match even years, so round up the years
+                # pandas round function does not work for years, so do it manually
+                yr = date.dt.year.values
+                mo = date.dt.month.values
+                dy = date.dt.day.values
+                dec_yr = np.around(yr + (30 * (mo - 1) + dy) / 365.0)  # approximate ok
+                date = f"{dec_yr.astype(int)}-01-01_00:00:00".ljust(64)
+                xtime.append(date)
+
+            ds["xtime"] = ("Time", xtime)
+            ds["xtime"] = ds.xtime.astype('S')
+            ds = ds.drop_vars(["Time"])
+
+        ismip6_to_mali_vars = dict(
+            thermal_forcing="ismip6shelfMelt_3dThermalForcing")
+        ds = ds.rename(ismip6_to_mali_vars)
+
+        # drop unnecessary variables
+        ds = ds.drop_vars(["z_bnds", "lat_vertices", "area",
+                           "lon_vertices", "lat", "lon"])
+
+        # transpose dimension
+        ds["thermal_forcing"] = ds["thermal_forcing"].transpose(
```

Comment:
This needs to be `ismip6shelfMelt_3dThermalForcing` instead of `thermal_forcing`.